gobblin-core/src/main/java/gobblin/instrumented/fork/InstrumentedForkOperatorBase.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.instrumented.fork;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import com.google.common.io.Closer;
+
+import gobblin.configuration.WorkUnitState;
+import gobblin.fork.ForkOperator;
+import gobblin.instrumented.Instrumentable;
+import gobblin.instrumented.Instrumented;
+import gobblin.metrics.MetricContext;
+
+
+/**
+ * package-private implementation of instrumentation for {@link gobblin.fork.ForkOperator}.
+ * See {@link gobblin.instrumented.fork.InstrumentedForkOperator} for extensible class.
+ */
+abstract class InstrumentedForkOperatorBase<S, D> implements Instrumentable, ForkOperator<S, D> {
+
+  protected MetricContext metricContext = new MetricContext.Builder("TMP").build();
+  protected Closer closer = Closer.create();
+  // Initialize as dummy metrics to avoid null pointer exception if init was skipped
+  protected Meter inputMeter = new Meter();
+  protected Meter outputForks = new Meter();
+  protected Timer forkOperatorTimer = new Timer();
+
+  @Override
+  public void init(WorkUnitState workUnitState)
+      throws Exception {
+    this.metricContext = closer.register(Instrumented.getMetricContext(workUnitState, this.getClass()));
+
+    this.inputMeter = this.metricContext.meter("gobblin.fork.operator.records.in");
+    this.outputForks = this.metricContext.meter("gobblin.fork.operator.forks.out");
```

Comment:
It is interesting in terms of tracking where things might be going wrong. If a run of Gobblin is not producing all expected outputs, this metric will help us know if the problem is in fork operator logic, or somewhere downstream.